### PR TITLE
Adds a outline filter to synthetic protective and immobile forms.

### PR DESCRIPTION
**Pull request by carlarctg**
# About the pull request

Adds an outline filter to synthetic protective and immobile forms. Blue = Immobile, Green = Protective

The smartpack's visual-only exoskeleton has been set to default - but is now only visible when in repair, protective, or immobile forms. These sprites are preexisting so no sprite rewview please :/

# Explain why it's good for the game

> Adds a outline filter to synthetic protective and immobile forms. Slightly bigger if exoskeleton is disabled.

There's not really any way to notice a synthetic running around in protective form if they walk in from offscreen with it, or really in the middle of combat. To the average marine or xeno it just looks like they're rushing in, tanking 3 gorillion damage, and leaving. By adding a filter so their form is actually visible it will give feedback to everyone that it's active. Added it to immobile as well for completeness.

# Testing Photographs and Procedure

<details>
<summary>Screenshots & Videos</summary>

Put screenshots and videos here with an empty line between the screenshots and the `<details>` tags.

</details>

# Changelog

:cl:
balance: Adds an outline filter to synthetic protective and immobile forms. Blue = Immobile, Green = Protective
balance: The smartpack's visual-only exoskeleton has been set to default - but is now only visible when in repair, protective, or immobile forms. T
/:cl:

### CODE REVIEW

**Comment by carlarctg** [comment]
what is that detritus commit i thought i reset master :/

**Review by stanalbatross — CHANGES_REQUESTED** [review]
post screenshots of them in use

File: code/game/objects/items/storage/smartpack.dm [hunk 1]
```diff
@@ -43,8 +50,14 @@
 
 	show_exoskeleton = !show_exoskeleton
 	to_chat(usr, SPAN_NOTICE("\The [src] will [show_exoskeleton ? "now" : "no longer"] have an exoskeleton."))
+	playsound(src, 'sound/machines/click.ogg', 15, TRUE)
 	update_icon(usr)
 
+/obj/item/storage/backpack/marine/smartpack/clicked(mob/user, list/mods)
+	if(mods["ctrl"])
+		toggle_exoskeleton()
+		return TRUE
```
Comment (stanalbatross) [inline code/game/objects/items/storage/smartpack.dm:59]: can_pickup check pls

File: code/game/objects/items/storage/smartpack.dm [hunk 2]
```diff
@@ -201,19 +216,26 @@
 	to_chat(user, SPAN_INFO("The current charge reads [battery_charge]/[SMARTPACK_MAX_POWER_STORED]"))
 	update_icon(user)
 
+	var/filter_size = EXOSKELETON_OFF_FILTER_SIZE
+	var/filter_color = PROTECTIVE_FORM_COLOR
+	if(show_exoskeleton)
+		filter_size = EXOSKELETON_ON_FILTER_SIZE
+	user.add_filter("synth_protective_form", 1, list("type" = "outline", "color" = filter_color, "size" = filter_size))
```
Comment (stanalbatross) [inline code/game/objects/items/storage/smartpack.dm:223]: named args for 1 and the list

File: code/game/objects/items/storage/smartpack.dm [hunk 3]
```diff
@@ -232,11 +254,18 @@
 		user.frozen = TRUE
 		to_chat(user, SPAN_DANGER("[name] beeps, \"You are anchored in place and cannot be moved.\""))
 		to_chat(user, SPAN_INFO("The current charge reads [battery_charge]/[SMARTPACK_MAX_POWER_STORED]"))
+
+		var/filter_size = EXOSKELETON_OFF_FILTER_SIZE
+		var/filter_color = IMMOBILE_FORM_COLOR
+		if(show_exoskeleton)
+			filter_size = EXOSKELETON_ON_FILTER_SIZE
+		user.add_filter("synth_immobile_form", 1, list("type" = "outline", "color" = filter_color, "size" = filter_size))
```
Comment (stanalbatross) [inline code/game/objects/items/storage/smartpack.dm:262]: same

File: code/game/objects/items/storage/smartpack.dm [hunk 4]
```diff
@@ -14,14 +20,15 @@
...
-	var/show_exoskeleton = FALSE
+	var/show_exoskeleton = TRUE
 
 	var/flashlight_cooldown = 0 //Cooldown for toggling the light
 	var/light_state = FALSE //Is the light on or off
 
 	var/battery_charge = SMARTPACK_MAX_POWER_STORED //How much power are we storing
 	var/activated_form = FALSE
 	var/immobile_form = FALSE
+	var/repair_form = FALSE
```
Comment (stanalbatross) [inline code/game/objects/items/storage/smartpack.dm:31]: you could make all these into flags/a single var with defines... i wont force you... but you could...

**Review by stanalbatross — CHANGES_REQUESTED** [review]

File: code/game/objects/items/storage/smartpack.dm (hunk 2 above)
Comment (carlarctg) [reply code/game/objects/items/storage/smartpack.dm:223]: bruh thats nowhere else in code!!!!!!!!!!!! how inconsistent!!!!!!!!!!!!!!

File: code/game/objects/items/storage/smartpack.dm (hunk 4 above)
Comment (carlarctg) [reply code/game/objects/items/storage/smartpack.dm:31]: nyoooo

File: code/game/objects/items/storage/smartpack.dm (hunk 1 above)
Comment (carlarctg) [reply code/game/objects/items/storage/smartpack.dm:59]: yea

File: code/game/objects/items/storage/smartpack.dm (hunk 2 above)
Comment (stanalbatross) [reply code/game/objects/items/storage/smartpack.dm]: but it's in standards and styles

File: code/game/objects/items/storage/smartpack.dm [hunk 5]
```diff
@@ -201,19 +216,27 @@
 	to_chat(user, SPAN_INFO("The current charge reads [battery_charge]/[SMARTPACK_MAX_POWER_STORED]"))
 	update_icon(user)
 
+	var/filter_color = PROTECTIVE_FORM_COLOR
+	var/filter_size = EXOSKELETON_OFF_FILTER_SIZE
+	if(show_exoskeleton)
+		filter_size = EXOSKELETON_ON_FILTER_SIZE
+	var/filter_priority = 1
+	user.add_filter("synth_protective_form", filter_priority, list("type" = "outline", "color" = filter_color, "size" = filter_size))
```
Comment (stanalbatross) [inline code/game/objects/items/storage/smartpack.dm:224]: this isn't how you do named args
it's in the format proc(argname = argvalue, argname2 = argvalue2)

File: code/game/objects/items/storage/smartpack.dm [hunk 6]
```diff
@@ -232,11 +255,19 @@
 		user.frozen = TRUE
 		to_chat(user, SPAN_DANGER("[name] beeps, \"You are anchored in place and cannot be moved.\""))
 		to_chat(user, SPAN_INFO("The current charge reads [battery_charge]/[SMARTPACK_MAX_POWER_STORED]"))
+
+		var/filter_color = IMMOBILE_FORM_COLOR
+		var/filter_size = EXOSKELETON_OFF_FILTER_SIZE
+		if(show_exoskeleton)
+			filter_size = EXOSKELETON_ON_FILTER_SIZE
+		var/filter_priority = 1
```
Comment (stanalbatross) [inline code/game/objects/items/storage/smartpack.dm:263]: 💀

**Review by stanalbatross — CHANGES_REQUESTED** [review]

File: code/game/objects/items/storage/smartpack.dm (hunk 6 above)
Comment (carlarctg) [reply code/game/objects/items/storage/smartpack.dm]: i hate you!!

File: code/game/objects/items/storage/smartpack.dm (hunk 5 above)
Comment (carlarctg) [reply code/game/objects/items/storage/smartpack.dm]: wah

**Review by harryob — APPROVED** [review]
the code is fine (and it is architect approved, so!)

**Pull request merged**